signer/local_cosigner.go

```diff
@@ -419,30 +420,45 @@ func (cosigner *LocalCosigner) GetEphemeralSecretParts(
 	}
 
 	res := &CosignerEphemeralSecretPartsResponse{
-		EncryptedSecrets: make([]CosignerEphemeralSecretPart, 0, len(cosigner.rsaPubKeys)-1),
+		EncryptedSecrets: make([]CosignerEphemeralSecretPart, len(cosigner.rsaPubKeys)-1),
 	}
 
 	id := cosigner.GetID()
 
-	for _, pubKey := range cosigner.rsaPubKeys {
+	var eg errgroup.Group
+
+	for i, pubKey := range cosigner.rsaPubKeys {
 		if pubKey.ID == id {
 			continue
 		}
-		secretPart, err := cosigner.getEphemeralSecretPart(CosignerGetEphemeralSecretPartRequest{
-			ChainID:   chainID,
-			ID:        pubKey.ID,
-			Height:    hrst.Height,
-			Round:     hrst.Round,
-			Step:      hrst.Step,
-			Timestamp: time.Unix(0, hrst.Timestamp),
-		})
 
-		if err != nil {
-			return nil, err
-		}
+		i := i
+		id := pubKey.ID
+
+		eg.Go(func() error {
+			secretPart, err := cosigner.getEphemeralSecretPart(CosignerGetEphemeralSecretPartRequest{
+				ChainID:   chainID,
+				ID:        id,
+				Height:    hrst.Height,
+				Round:     hrst.Round,
+				Step:      hrst.Step,
+				Timestamp: time.Unix(0, hrst.Timestamp),
+			})
+
+			if i > id {
+				res.EncryptedSecrets[i-2] = secretPart
+			} else {
+				res.EncryptedSecrets[i-1] = secretPart
+			}
```

Comment:
This threw me off for a moment since it wasn't immediately clear that `i` was map keys -- it seemed that there was an unhandled `i == 0`  case that would lead to indexing a slice at -1.

It could maybe be clearer with `idx := i-1` instead of `i := i`, but this is narrow enough in scope that I don't think it really needs to change.

Reply:
good call, updated